test/lib/pgBackRestTest/Module/Manifest/ManifestAllTest.pm

```diff
@@ -0,0 +1,1088 @@
+
+        # Undefined user/group
+        #---------------------------------------------------------------------------------------------------------------------------
+        executeTest("sudo chgrp 777 " . $self->{strDbPath} . '/pg_xlog/' . BOGUS);
+        executeTest("sudo chown 777 " . $self->{strDbPath} . '/pg_xlog/' . BOGUS);
+        $oManifestExpected->boolSet(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/pg_xlog/' . BOGUS,
+            MANIFEST_SUBKEY_GROUP, false);
+        $oManifestExpected->boolSet(MANIFEST_SECTION_TARGET_FILE, MANIFEST_TARGET_PGDATA . '/pg_xlog/' . BOGUS,
+            MANIFEST_SUBKEY_USER, false);
+
+        $oManifest->build(storageDb(), $self->{strDbPath}, undef, false);
+        $self->testResult(sub {$self->manifestCompare($oManifestExpected, $oManifest)}, "",
+            'undefined user/group');
+
+        # Reset the group / owner
+        executeTest("sudo chgrp " . TEST_GROUP . " " . $self->{strDbPath} . '/pg_xlog/' . BOGUS);
+        executeTest("sudo chown " . TEST_USER . " " . $self->{strDbPath} . '/pg_xlog/' . BOGUS);
+    }
+
+    ################################################################################################################################
+    if ($self->begin('get/set'))
+    {
+        my $oManifest = new pgBackRest::Manifest($strBackupManifestFile, {bLoad => false, strDbVersion => PG_VERSION_94});
+
+        # SubKey required but has not been set
+        #---------------------------------------------------------------------------------------------------------------------------
+        $self->testException(sub {$oManifest->boolGet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE)},
+            ERROR_ASSERT, "strSection '" . MANIFEST_SECTION_TARGET_FILE . "', strKey '" . BOGUS . "', strSubKey '" .
+            MANIFEST_SUBKEY_CHECKSUM_PAGE . "' is required but not defined");
+
+        # SubKey not required and not set
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG Why would we allow this?
+        $self->testResult(sub {$oManifest->boolGet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE, false)},
+            false, 'boolGet() - false');
+
+        # Set and get a boolean value
+        #---------------------------------------------------------------------------------------------------------------------------
+        $oManifest->boolSet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE, true);
+        $self->testResult(sub {$oManifest->boolGet(MANIFEST_SECTION_TARGET_FILE, BOGUS, MANIFEST_SUBKEY_CHECKSUM_PAGE)}, true,
+            'boolGet() - true');
+
+        # Get default boolean value
+        #---------------------------------------------------------------------------------------------------------------------------
+# CSHANG But this doesn't actually set the default value, so should we be allowing this? Why?
```

Comment:
This is a different kind of default.  There are defaults for certain values in the manifest, and the Ini object also has the option to return a default if the value is not set (in that case the default will be provided in the call).